kafka/adminapi.go

```diff
@@ -2197,8 +2198,12 @@ func (a *AdminClient) ListConsumerGroupOffsets(
 	for i, groupPartitions := range groupsPartitions {
 		// We need to destroy this list because rd_kafka_ListConsumerGroupOffsets_new
 		// creates a copy of it.
-		cPartitions := newCPartsFromTopicPartitions(groupPartitions.Partitions)
-		defer C.rd_kafka_topic_partition_list_destroy(cPartitions)
+		var cPartitions *C.rd_kafka_topic_partition_list_t = nil
+
+		if groupPartitions.Partitions != nil {
```

Comment:
I think we should allow both or non.

In Python, we are not allowing empty list as well for partitions. Let's check what is that we really want to support.

Reply:
We are allowing both from this PR in the Go implementation.
librdkafka is handling the actual meaning of each kind of list, we're just letting it pass through. If we want to change the meaning of each type, we should be making that change in librdkafka, which wouldn't be possible anymore since it will change the API.
Earlier, we were not passing the nil list correctly to librdkafka, that is the bug.

filled list -> get those topic partitions only
nil list -> get all topic partitions
empty list -> local error.

I don't think we should do any error handling at level of the binding unless it's something binding specific, and we should just let the librdkafka error bubble up.